src/main/java/com/ramcel/cinema/reservation/functionalities/ticket/validators/TicketValidator.java

```diff
@@ -0,0 +1,184 @@
+package com.ramcel.cinema.reservation.functionalities.ticket.validators;
+
+import com.ramcel.cinema.reservation.db.entity.SeatEntity;
+import com.ramcel.cinema.reservation.db.repositories.SeatRepository;
+import com.ramcel.cinema.reservation.functionalities.exception.IllegalReservationException;
+import com.ramcel.cinema.reservation.functionalities.exception.IllegalSeatException;
+import com.ramcel.cinema.reservation.functionalities.exception.IllegalTicketException;
+import com.ramcel.cinema.reservation.functionalities.seat.SeatStatus;
+import com.ramcel.cinema.reservation.functionalities.ticket.Ticket;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.time.LocalDateTime;
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Component
+public class TicketValidator {
+
+    @Autowired
+    private SeatRepository seatRepository;
```

Comment:
Czy `Validator` powinien mieć dostęp do bazy danych? :thinking: Nie wiem jak to zaplanowałeś, może nazwa mnie myli ale `Validator` brzmi jak coś co sprawdza poprawność dostarczonych danych. Dodanie tutaj zależności wydaje się komplikować architekturę oraz testowanie, by przetestować `Validator` musisz np. mockować bazę danych. 

Ciekawi mnie jak Ty to widzisz :slightly_smiling_face:

Reply:
Po fakcie naszło mnie już, że tę klasę możnaby rozbić na dwie - jakiś BasicValidator i RequirementsValidator (czy coś takiego, nazwa robocza). Ta klasa ogólnie trochę urosła, BasicValidator mógłby się zająć sprawdzaniem wymagań, które nie potrzebują danych z bazy (długość imienia itd). Wtedy byłaby to w zasadzie w pełni statyczna, funkcyjna klasa (funkcyjna klasa?).  Ten drugi potrzebuje dane z bazy, ponieważ business requirementy są takie, że nie możemy rezerwować miejsc, w taki sposób, żeby zostały dziury. 

Zdecydowałem się na rozwiązanie tego w walidacji, bo z mojego punktu widzenia to jest nieważny bilet. Drugim rozwiązaniem byłoby puszczenie przez walidację tylko rzeczy, które możemy sprawdzić bez bazy i sprawdzanie, czy nasza rezerwacja się nie wywala przy momencie przeprowadzania samej rezerwacji. No ale to jest separation of concerns i trochę syf moim zdaniem.